### PR TITLE
Updates to serialization.md

### PR DESCRIPTION
**Pull request by suo**
Stack from [ghstack](https://github.com/ezyang/ghstack):
* #31373 Update OVERVIEW.md
* **#31372 Updates to serialization.md**

Keeping it current with the latest changes.

Differential Revision: [D19145986](https://our.internmc.facebook.com/intern/diff/D19145986)

### CODE REVIEW

**Comment by kostmo** [comment]
## CircleCI build failures summary
As of commit 6f58a6b:
* **1/1** failures introduced in this PR
* **1/1** recognized as flaky
    * Re-run these jobs?



## Detailed failure analysis

One may explore the probable reasons each build failed interactively [on the Dr. CI website](https://dr.pytorch.org/commit-details.html?sha1=6f58a6b2a2b1f0b63d4cba515b5283a6fcabf83d).

### 1 new failure recognized by patterns
The following build failures don't appear to be due to upstream breakage:
#### [See CircleCI build](https://circleci.com/gh/pytorch/pytorch/3982130) pytorch_xla_linux_xenial_py3_6_clang7_test (1/1)
**Step:** "Set Up CI Environment After attach_workspace" ([full log](https://dr.pytorch.org/api/view-log-full?build_id=75986305) | [pattern match details](https://dr.pytorch.org/build-details.html?build_id=75986305))
<details>
<summary>
<code>E: Failed to fetch http://us-central1.gce.archive.ubuntu.com/ubuntu/dists/xenial-updates/main/binary-amd64/Packages.xz  Hash Sum mismatch</code>
</summary>

```
                                                              89% [27 Packages store 0 B]                             Get:57 http://us-central1.gce.archive.ubuntu.com/ubuntu xenial-backports/main amd64 Packages [7,280 B] 
                                                              89% [27 Packages store 0 B]                             Get:58 http://us-central1.gce.archive.ubuntu.com/ubuntu xenial-backports/main Translation-en [4,456 B] 
                                                                     89% [27 Packages store 0 B]                             Get:59 http://us-central1.gce.archive.ubuntu.com/ubuntu xenial-backports/universe amd64 Packages [8,064 B] 
                                                               89% [27 Packages store 0 B]                             Get:60 http://us-central1.gce.archive.ubuntu.com/ubuntu xenial-backports/universe Translation-en [4,328 B] 
                                                                     89% [27 Packages store 0 B]                             Get:47 http://us-central1.gce.archive.ubuntu.com/ubuntu xenial-updates/main amd64 Packages [1,082 kB] 
 89% [27 Packages store 0 B] [47 Packages 27.8 kB/1,082 kB 3%]                                                               Err:47 http://us-central1.gce.archive.ubuntu.com/ubuntu xenial-updates/main amd64 Packages 
  Hash Sum mismatch 
                                                              91% [27 Packages store 0 B]                             Get:51 http://us-central1.gce.archive.ubuntu.com/ubuntu xenial-updates/universe amd64 Packages [771 kB] 
                                  95% [Working]               Fetched 28.8 MB in 5s (5,381 kB/s) 
Reading package lists... 98%  Reading package lists... Done  
E: Failed to fetch http://us-central1.gce.archive.ubuntu.com/ubuntu/dists/xenial-updates/main/binary-amd64/Packages.xz  Hash Sum mismatch 
E: Some index files failed to download. They have been ignored, or old ones used instead. 
```

</details>



---

<details><summary>This comment was automatically generated by <a href="https://github.com/kostmo/circleci-failure-tracker/tree/master/docs/from-pull-request-comment">Dr. CI</a> (expand for details).</summary>Follow <a href="https://dr.pytorch.org/admin/comments-opt-out.html">this link to opt-out</a> of these comments for your Pull Requests.

Please report bugs/suggestions on the <a href="https://github.com/kostmo/circleci-failure-tracker/issues">GitHub issue tracker</a>.
</details>

File: torch/csrc/jit/docs/serialization.md [hunk 1]
```diff
@@ -278,30 +236,52 @@ necessary for pickling a module object.
...
-3. Unpickle `data.pkl` into the top-level `script::Module` and return it
-
-The first two steps will go away soon, so this section will focus on the
-unpickling.
+1. Unpickle `constants.pkl`, which produces a tuple of all tensor constants
+   referenced in code.
+2. Unpickle `data.pkl` into the top-level `script::Module` and return it.
 
 The unpickling process consists of a single call to unpickle the module
 object contained `data.pkl`. The `Unpickler` is given a callback that lets it
```
Comment (bhosmer) [inline torch/csrc/jit/docs/serialization.md]: contained in `data.pkl', since we're in here :)

**Review by bhosmer — APPROVED** [review]
These updates are actually really useful. Couple of things inline, use at your discretion :)

File: torch/csrc/jit/docs/serialization.md [hunk 2]
```diff
@@ -278,30 +236,52 @@ necessary for pickling a module object.
...
-(see "tensors" below). PyTorch functions defined in
-[torch/jit/_pickle.py](../../../jit/_pickle.py) used to mark special data
-types, such as this tensor table index or specialized lists.
+(see [the tensor section](#tensors-How-tensors-are-serialized) below).
+"Data" means all parts of the module object state, like attributes,
+submodules, etc.
 
-## `tensors/`: How tensors are serialized
+PyTorch functions defined in [torch/jit/_pickle.py](../../../jit/_pickle.py)
+used to mark special data types, such as this tensor table index or
```
[the remark this answers is not in the document]
Comment (suo) [reply torch/csrc/jit/docs/serialization.md]: good catch! there is a missing "are"

**Pull request closed without merging**